src/routes/webhooks/paddle.ts

```diff
@@ -192,68 +180,93 @@ const planChanged = async (data: SubscriptionUpdatedEvent) => {
 
   return (
     (flags?.subscriptionFlags?.cycle as string) !==
-    extractSubscriptionType(data.data?.items)
+    extractSubscriptionCycle(data?.items)
   );
 };
 
+interface AnalyticsExtra {
+  cycle: string;
+  cost: number;
+  currency: string;
+  payment: string;
+  localCost: number;
+  localCurrency: string;
+  payout: TransactionPayoutTotalsNotification | null;
+}
+
+const getAnalyticsExtra = (
+  data: (
+    | SubscriptionUpdatedEvent
+    | SubscriptionCanceledEvent
+    | TransactionCompletedEvent
+  )['data'],
+): Partial<AnalyticsExtra> => {
+  const cost = data.items?.[0]?.price?.unitPrice?.amount;
+  const currency = data.items?.[0]?.price?.unitPrice?.currencyCode;
+  const localCurrency = data.currencyCode;
+
+  // payments are only available on transaction events
+  if (!('payments' in data)) {
+    return {
+      cycle: extractSubscriptionCycle(data.items),
+      cost: cost ? parseInt(cost) / 100 : undefined,
+      currency,
+      localCurrency,
+    };
+  }
+
+  const transaction = data as TransactionCompletedEvent['data'];
+  const localCost = transaction?.details?.totals?.total;
+  const payout = transaction?.details?.payoutTotals;
+  const payment = transaction.payments?.reduce((acc, item) => {
+    if (item.status === 'captured') {
+      acc = item?.methodDetails?.type || '';
+    }
+    return acc;
+  }, '');
+
+  return {
+    cost: cost ? parseInt(cost) / 100 : undefined,
+    currency,
+    payment,
+    localCost: localCost ? parseInt(localCost) / 100 : undefined,
+    localCurrency,
+    payout,
+  };
+};
```

Comment:
Basically took the deleted block of codes below to be moved here. Due to the following concerns:
- Typescript issue.
- A lot of places were using `as Type` which can be done just once if as displayed above.
- It was a bit hard to follow which properties are expected to have values and which ones are not.